packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -2039,87 +2039,20 @@ class _VerticalOuterDimension extends Scrollable {
   _VerticalOuterDimensionState createState() => _VerticalOuterDimensionState();
 }
 
-class _VerticalOuterDimensionState extends ScrollableState {
-  DiagonalDragBehavior get diagonalDragBehavior => (widget as _VerticalOuterDimension).diagonalDragBehavior;
-
-  @override
-  void setCanDrag(bool value) {
-    switch (diagonalDragBehavior) {
-      case DiagonalDragBehavior.none:
-        // If we aren't scrolling diagonally, the default drag gesture
-        // recognizer is used.
-        super.setCanDrag(value);
-        return;
-      case DiagonalDragBehavior.weightedEvent:
-      case DiagonalDragBehavior.weightedContinuous:
-      case DiagonalDragBehavior.free:
-        if (value) {
-          // If a type of diagonal scrolling is enabled, a panning gesture
-          // recognizer will be created for the _InnerDimension. So in this
-          // case, the _OuterDimension does not require a gesture recognizer.
-          _gestureRecognizers = const <Type, GestureRecognizerFactory>{};
-          // Cancel the active hold/drag (if any) because the gesture recognizers
-          // will soon be disposed by our RawGestureDetector, and we won't be
-          // receiving pointer up events to cancel the hold/drag.
-          _handleDragCancel();
-          _lastCanDrag = value;
-          _lastAxisDirection = widget.axis;
-          if (_gestureDetectorKey.currentState != null) {
-            _gestureDetectorKey.currentState!.replaceGestureRecognizers(_gestureRecognizers);
-          }
-        }
-        return;
-    }
-  }
-
-  @override
-  Widget _buildChrome(BuildContext context, Widget child) {
-    final ScrollableDetails details = ScrollableDetails(
-      direction: widget.axisDirection,
-      controller: _effectiveScrollController,
-      clipBehavior: widget.clipBehavior,
-    );
-    // Skip building a scrollbar here, the dual scrollbar is added in
-    // TwoDimensionalScrollableState.
-    return _configuration.buildOverscrollIndicator(context, child, details);
-  }
+class _TwoDimensionalDragForwarding {
```

Comment:
Public docs are not required for private classes (`///`), but can you add some docs to this for maintainers to better understand what it's for and where ti is used?